### DISCUSSION TITLE
Put document question

### DISCUSSION THREAD

**Discussion by link4531** [TomWright/dasel#132] [Q&A] [answered]
Working off the 'put document' examples provided in the documentation 

lets say I have a
doc.yaml 
> ```shell
> name: tom
> color: red
> ```

Running the command:
`echo '{"people":[]}' | dasel put document -p json -d yaml '.people.[]' $(cat doc.yaml)`

returns 
>```shell
>{
>  "people": [
>    {
>      "name": null
>    }
>  ]
>}
>```

however if I run the command
> ```shell 
>echo '{"people":[]}' | dasel put document -p json -d yaml '.people.[]' 'name: tom
> color: red' 
>```
everything runs as you would expect...

dasel 
- begins to parse the file
- it accepts the format of doc.yaml being yaml
- dasel spits out a null. 

Questions
is this intended? am I being dumb?
does this functionality require the features from the 'document merge' branch?

**Comment by TomWright** [comment, marked as answer]
You need to wrap the cat argument in double quotes so as it's picked up as a single string:

```bash
$ echo '{"people":[]}' | dasel put document -p json -d yaml '.people.[]' "$(cat doc.yaml)"
{
  "people": [
    {
      "color": "red",
      "name": "tom"
    }
  ]
}
```

In your example, dasel is essentially given a null value which is why you got that result.

**Reply by link4531** [reply]
... oh man. I knew I was doing something dumb... But your answer makes complete sense. Thank you
